ts/Extensions/ExportData.ts

```diff
@@ -469,6 +471,71 @@ addEvent(Chart, 'render', function (): void {
     }
 });
 
+addEvent(Chart, 'afterViewData', function (): void {
+    const chart = this,
+        dataTableDiv = chart.dataTableDiv,
+        row = document.getElementsByClassName('highcharts-table-sort-row')[0],
+        getCellValue = function (
+            tr: HTMLDOMElement,
+            index: number
+        ): string|null {
+            return tr.children[index].textContent;
+        },
+        comparer = function (index: number, ascending: boolean) {
+            return function (a: HTMLDOMElement, b: HTMLDOMElement): number {
+                const sort = function (v1: any, v2: any): number {
+                    return v1 !== '' && v2 !== '' && !isNaN(v1) && !isNaN(v2) ?
+                        v1 - v2 :
+                        v1.toString().localeCompare(v2);
+                };
+
+                return sort(
+                    getCellValue(ascending ? a : b, index),
+                    getCellValue(ascending ? b : a, index)
+                );
+            };
+        };
+
+    if (dataTableDiv) {
+        row.childNodes.forEach((th: any): void => {
+            const table = th.closest('table');
+
+            th.addEventListener('click', function (): void {
+                const thNodeList = dataTableDiv.querySelectorAll(
+                        'tr:not(.highcharts-table-sort-row)'
+                    ),
+                    thParentNodeList = th.parentNode.children,
+                    thArrayList = [] as Array<HTMLElement>,
+                    thParentArray = [] as Array<HTMLElement>;
+
+
+                [].forEach.call(
```

Comment:
Are you just copying into an array? Then you can use destructuring:

`thArrayList = [...thNodeList]`

Reply:
I'm trying to convert `NodeListOf<Element>` into a proper array that I can letter use the Array `sort` method.

I can destruct it as you suggested but it won't be possible without some strange casting but I'm ok with that.
This approach is much more readable than this odd `[].forEach` that I used.